### PR TITLE
406122: Public Keyvault

### PR DESCRIPTION
**Pull request by rajeshkaremane**
# **What this PR does / why we need it**:
*A brief description of changes being made*
*Link to the relevant work items: *
[AB#406122](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/406122) 

# **Special notes for your reviewer**
*Any specific actions or notes on review?*

# Testing
*Any relevant testing information and pipeline runs.*
https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=604415&view=results

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [x] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: infra/core/env/key-vault-public/key-vault-public.bicep [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    tags: union(defaultTags, keyVaultTags)
+    vaultSku: keyVault.skuName
+    lock: 'CanNotDelete'
+    enableRbacAuthorization: true
+    enableSoftDelete: bool(keyVault.enableSoftDelete)
+    enablePurgeProtection: bool(keyVault.enablePurgeProtection)
+    softDeleteRetentionInDays: int(keyVault.softDeleteRetentionInDays)
+    networkAcls: {
+      bypass: 'AzureServices'
+      defaultAction: 'Deny'
```
Comment (DaRoszko) [inline infra/core/env/key-vault-public/key-vault-public.bicep:83]: is this file needed? i.e. can't we just have a parameter instantiation of the existing keyvault.main file?

Comment (rajeshkaremane) [reply infra/core/env/key-vault-public/key-vault-public.bicep:83]: It is possible, I have created separate PR: https://github.com/DEFRA/adp-infrastructure-core/pull/218
